Author: Luca Boccassi <luca.boccassi@microsoft.com>
Description: Automake: use EXTRA_LTLIBRARIES instead of noinst_LTLIBRARIES
 noinst_LTLIBRARIES causes the libraries to be always built
 unconditionally. EXTRA_LTLIBRARIES causes them to be built
 only if other build target needs them.
 In other words, avoid building libcommon.a and libtcolors.a
 unless they are needed by another library/executable and
 save some build time.
Upstream-Status: backport, commit:c65953d72bbc7412f32e566d9fa6e780d84f0696
--- a/Makefile.am
+++ b/Makefile.am
@@ -39,7 +39,7 @@ bashcompletiondir = @bashcompletiondir@
 
 dist_noinst_HEADERS =
 noinst_PROGRAMS =
-noinst_LTLIBRARIES =
+EXTRA_LTLIBRARIES =
 usrbin_exec_PROGRAMS =
 usrsbin_exec_PROGRAMS =
 dist_man_MANS =
@@ -169,7 +169,7 @@ else
 edit_cmd += -e 's|@vendordir[@]||g'
 endif
 
-CLEANFILES += $(PATHFILES)
+CLEANFILES += $(PATHFILES) $(EXTRA_LTLIBRARIES)
 EXTRA_DIST += $(PATHFILES:=.in)
 
 $(PATHFILES): Makefile
--- a/lib/Makemodule.am
+++ b/lib/Makemodule.am
@@ -9,7 +9,7 @@
 # Note that you need "make install-strip" (or proper rpm / Debian build)
 # to generate binaries with only relevant stuff.
 #
-noinst_LTLIBRARIES += libcommon.la
+EXTRA_LTLIBRARIES += libcommon.la
 libcommon_la_CFLAGS = $(AM_CFLAGS)
 libcommon_la_SOURCES = \
 	lib/blkdev.c \
@@ -59,7 +59,7 @@ libcommon_la_SOURCES += lib/sysfs.c
 endif
 endif
 
-noinst_LTLIBRARIES += libtcolors.la
+EXTRA_LTLIBRARIES += libtcolors.la
 libtcolors_la_CFLAGS = $(AM_CFLAGS)
 libtcolors_la_SOURCES = lib/colors.c lib/color-names.c include/colors.h include/color-names.h
 libtcolors_la_LIBADD =
